src/cpp/session/modules/SessionDataViewer.R

```diff
@@ -132,7 +132,13 @@
       if (is.character(label))
          col_label <- label
       else if (idx <= length(colLabels))
+      {
+      # check if colLabels is a named vector, if yes, use the named value, if not use the idx'th element
+      if (!is.null(names(colLabels)))
+         col_label <- colLabels[[col_name]]
```

Comment:
We should verify that `col_name` is actually a name that exists within `names(colLabels)`.